src/App.tsx

```diff
@@ -1,24 +1,232 @@
-/* eslint-disable max-len */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useState, useEffect, useCallback } from 'react';
+import classNames from 'classnames';
 import { UserWarning } from './UserWarning';
+import { Todo } from './types/Todo';
+import { MainFilter } from './types/MainFilter';
+import {
+  getTodos, addTodo, deleteTodo, updateTodo,
+} from './api/todos';
+import { TodoList } from './components/TodoList/TodoList';
+import { TodoFilter } from './components/TodoFilter/TodoFilter';
+import { Loader } from './components/Loader/Loader';
+import { TodoError } from './components/TodoError/TodoError';
 
-const USER_ID = 0;
+const USER_ID = 9975;
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [errorOfUpdate, setErrorOfUpdate] = useState('');
+  const [isTodosLoading, setIsTodosLoading] = useState(true);
+  const [filterOption, setFilterOption] = useState(MainFilter.All);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [textOfInput, setTextOfInput] = useState('');
+  const [loadingIds, setLoadingIds] = useState<number[]>([]);
+
+  const numberOfActiveTodos = todos
+    .filter(({ completed }) => !completed).length;
+
+  const numberOfCompletedTodos = todos
+    .filter(({ completed }) => completed).length;
+
+  const isInputDisabled = !!tempTodo && !!loadingIds.length;
+
+  const getTodosFromServer = async () => {
+    try {
+      const todosFromServer = await getTodos(USER_ID);
+
+      setTodos(todosFromServer);
+    } catch {
+      setErrorOfUpdate('Something was wrong, during the loading...');
+    } finally {
+      setIsTodosLoading(false);
+    }
+  };
+
+  useEffect(() => {
+    setErrorOfUpdate('');
+    getTodosFromServer();
+    setTimeout(() => setErrorOfUpdate(''), 3000);
+  }, []);
+
+  const getFilteredTodos = (filter: MainFilter, someTodos: Todo[]) => {
+    switch (filter) {
+      case MainFilter.Active:
+        return someTodos.filter((todo) => !todo.completed);
+      case MainFilter.Completed:
+        return someTodos.filter((todo) => todo.completed);
+      default:
+        return someTodos;
+    }
+  };
+
+  const addNewTodo = async (title: string) => {
+    const newTodo = {
+      title,
+      userId: USER_ID,
+      completed: false,
+    };
+
+    if (title.trim() === '') {
+      setErrorOfUpdate('Title can\'t be empty');
+      setTextOfInput('');
+      setTimeout(() => {
+        setErrorOfUpdate('');
+      }, 3000);
```

Comment:
you are doing it (resetting error inside timeout) a lot of times. I recommend you handle it inside `TodoError` component in `useEffect` hook (with the error in the dependency array) - if error is not empty reset it in timer as you do it here